### PR TITLE
TAJO-1877: Adopt try-resource statement to AbstractDBStore

### PR DESCRIPTION
**Pull request by dkhwangbo**
See https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html. 
JDK 7 introduced try-resource statement for better resource allocation and deallocation. 
We can also adopt this feature to AbstractDBStore.

### CODE REVIEW

**Comment by dkhwangbo** [comment]
@jihoonson I resolve some problem about rebasing my prev-work and create PR again.
Travis-ci gives me the error message but It's not occured in my local environment.
Can you run test of this PR in your local environment?

File: tajo-catalog/tajo-catalog-server/src/main/java/org/apache/tajo/catalog/store/AbstractDBStore.java [hunk 1]
```diff
@@ -249,18 +246,12 @@ private int getSchemaVersion() {
       LOG.debug(sql.toString());
     }
 
-    try {
-      conn = getConnection();
-      pstmt = conn.prepareStatement(sql);
-      result = pstmt.executeQuery();
-
+    try (ResultSet result = getConnection().prepareStatement(sql).executeQuery()) {
```
Comment (jihoonson) [inline tajo-catalog/tajo-catalog-server/src/main/java/org/apache/tajo/catalog/store/AbstractDBStore.java]: PreparedStatement is not closed.

File: tajo-catalog/tajo-catalog-server/src/main/java/org/apache/tajo/catalog/store/AbstractDBStore.java [hunk 2]
```diff
@@ -427,29 +408,20 @@ public void dropTablespace(String tableSpaceName) throws UndefinedTablespaceExce
...
-        sql += " WHERE " + whereCondition;
-      }
+    if (whereCondition != null) {
+      sql += " WHERE " + whereCondition;
+    }
 
-      conn = getConnection();
-      pstmt = conn.prepareStatement(sql);
-      resultSet = pstmt.executeQuery();
+    try (ResultSet resultSet = getConnection().prepareStatement(sql).executeQuery()) {
```
Comment (jihoonson) [inline tajo-catalog/tajo-catalog-server/src/main/java/org/apache/tajo/catalog/store/AbstractDBStore.java]: PreparedStatement is not closed.

File: tajo-catalog/tajo-catalog-server/src/main/java/org/apache/tajo/catalog/store/AbstractDBStore.java [hunk 3]
```diff
@@ -457,17 +429,11 @@ public void dropTablespace(String tableSpaceName) throws UndefinedTablespaceExce
...
     List<TablespaceProto> tablespaces = TUtil.newList();
 
-    try {
-      String sql = "SELECT SPACE_ID, SPACE_NAME, SPACE_HANDLER, SPACE_URI FROM " + TB_SPACES ;
-      conn = getConnection();
-      stmt = conn.createStatement();
-      resultSet = stmt.executeQuery(sql);
+    String sql = "SELECT SPACE_ID, SPACE_NAME, SPACE_HANDLER, SPACE_URI FROM " + TB_SPACES ;
 
+    try (ResultSet resultSet = getConnection().createStatement().executeQuery(sql)) {
```
Comment (jihoonson) [inline tajo-catalog/tajo-catalog-server/src/main/java/org/apache/tajo/catalog/store/AbstractDBStore.java]: PreparedStatement is not closed.

File: tajo-catalog/tajo-catalog-server/src/main/java/org/apache/tajo/catalog/store/AbstractDBStore.java [hunk 4]
```diff
@@ -678,29 +635,20 @@ public void dropDatabase(String databaseName) throws UndefinedDatabaseException
...
-        sql += " WHERE " + whereCondition;
-      }
+    if (whereCondition != null) {
+      sql += " WHERE " + whereCondition;
+    }
 
-      conn = getConnection();
-      pstmt = conn.prepareStatement(sql);
-      resultSet = pstmt.executeQuery();
+    try (ResultSet resultSet = getConnection().prepareStatement(sql).executeQuery()) {
```
Comment (jihoonson) [inline tajo-catalog/tajo-catalog-server/src/main/java/org/apache/tajo/catalog/store/AbstractDBStore.java]: PreparedStatement is not closed.

File: tajo-catalog/tajo-catalog-server/src/main/java/org/apache/tajo/catalog/store/AbstractDBStore.java [hunk 5]
```diff
@@ -708,18 +656,11 @@ public void dropDatabase(String databaseName) throws UndefinedDatabaseException
...
     List<DatabaseProto> databases = new ArrayList<>();
 
-    try {
-      String sql = "SELECT DB_ID, DB_NAME, SPACE_ID FROM " + TB_DATABASES;
+    String sql = "SELECT DB_ID, DB_NAME, SPACE_ID FROM " + TB_DATABASES;
 
-      conn = getConnection();
-      stmt = conn.createStatement();
-      resultSet = stmt.executeQuery(sql);
+    try (ResultSet resultSet = getConnection().createStatement().executeQuery(sql)) {
```
Comment (jihoonson) [inline tajo-catalog/tajo-catalog-server/src/main/java/org/apache/tajo/catalog/store/AbstractDBStore.java]: PreparedStatement is not closed.

**Pull request closed without merging**